lms/djangoapps/course_api/blocks/transformers/tests/test_milestones.py

```diff
@@ -171,3 +183,16 @@ def test_exam_created(self, exam_ref, attempt_status, expected_blocks):
             set(block_structure.get_block_keys()),
             set(self.get_block_key_set(self.blocks, *expected_blocks)),
         )
+
+    def test_exam_gated(self):
+        expected_blocks = ('course', 'A', 'B', 'C', 'TimedExam', 'D', 'E')
+        self.setup_gated_section(self.blocks['PracticeExam'], self.blocks['TimedExam'])
```

Comment:
There ought to be another test added for the case of content that requires milestones to be fulfilled, but is not a special (`timed/proctored/practice`) exam.